src/diagram/components/ui/variable-chip.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { observer } from "mobx-react-lite";
+import classNames from "classnames";
+
+import { VariableType } from "../../models/variable";
+
+import "./variable-chip.scss";
+
+export const kEmptyVariable = "<no name>";
+
+interface IVariableChip {
+  className?: string;
+  nameOnly?: boolean;
+  onClick?: (variable: VariableType) => void;
+  selected?: boolean;
+  variable: VariableType;
+}
+
+export const VariableChip = observer(({ className, nameOnly, onClick, selected, variable }: IVariableChip) => {
+  const name = variable.name;
+  const value = variable.computedValue;
+  const valueString = variable.computedValueWithSignificantDigits;
+  const unit = variable.computedUnit || "";
+  const showValue = value !== undefined && !isNaN(value);
```

Comment:
isNaN(undefined) returns true (I think) so maybe you don't need the first part?

Reply:
I tried removing the first part, and was surprised to get a type error. Apparently `isNaN()` only accepts numbers, which `NaN` technically is. Seems like it would be a lot more useful if it accepted `any` and just returned false for anything that isn't a number ¯\_(ツ)_/¯

Reply:
Interesting! Thanks for explaining!